src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -212,6 +212,103 @@ func migrateRepos(ctx context.Context, tx *pachsql.Tx) error {
 	return nil
 }
 
+// alterCommitsTable adds useful new columns to pfs.commits table.
+// Note that this is not the end all be all. We will need to make more changes after data has been migrated.
+// TODO
+//   - rename int_id to id? This will requires changing all references as well.
+//   - make repo_id not null
+//   - make origin not null
+//   - make updated_at not null and default to current timestamp
+//   - branch_id_str is a metadata reference to the branches table. Today this points to collections, but tomorrow it should
+//     point to pfs.branches. Once the PFS master watches branches, we will no longer need this column.
+func alterCommitsTable(ctx context.Context, tx *pachsql.Tx) error {
+	query := `
+	CREATE TYPE pfs.commit_origin AS ENUM ('ORIGIN_KIND_UNKNOWN', 'USER', 'AUTO', 'FSCK');
+
+	ALTER TABLE IF EXISTS pfs.commits
+		ADD COLUMN IF NOT EXISTS repo_id bigint REFERENCES pfs.repos(id),
+		ADD COLUMN IF NOT EXISTS origin pfs.commit_origin,
+		ADD COLUMN IF NOT EXISTS description text DEFAULT '',
+		ADD COLUMN IF NOT EXISTS start_time timestamptz,
+		ADD COLUMN IF NOT EXISTS finishing_time timestamptz,
+		ADD COLUMN IF NOT EXISTS finished_time timestamptz,
+		ADD COLUMN IF NOT EXISTS compacting_time bigint,
+		ADD COLUMN IF NOT EXISTS validating_time bigint,
+		ADD COLUMN IF NOT EXISTS error text,
+		ADD COLUMN IF NOT EXISTS size bigint,
+		ADD COLUMN IF NOT EXISTS updated_at timestamptz DEFAULT CURRENT_TIMESTAMP,
+		ADD COLUMN IF NOT EXISTS branch_id_str text;
```

Comment:
I thought this was going to be a BIGINT to the branches table?

Reply:
That's correct. I had implemented it as a string while waiting for Albert's schema changes on branches to be merged in (and the migration code as well).